client/state/data-layer/wpcom/domains/nameservers/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { DOMAIN_NAMESERVERS_FETCH, DOMAIN_NAMESERVERS_UPDATE } from 'calypso/state/action-types';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import {
+	fetchNameserversSuccess,
+	fetchNameserversFailure,
+	updateNameserversSuccess,
+} from 'calypso/state/domains/nameservers/actions';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+
+export const fetchDomainNameservers = ( action ) =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: `/domains/${ action.domainName }/nameservers/`,
+		},
+		action
+	);
+
+export const fetchDomainNameserversSuccess = ( { domainName }, nameservers ) =>
+	fetchNameserversSuccess( domainName, nameservers );
+
+export const fetchDomainNameserversError = ( { domainName } ) =>
+	fetchNameserversFailure( domainName );
+
+export const updateDomainNameservers = ( action ) =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: `/domains/${ action.domainName }/nameservers/`,
+			body: {
+				nameservers:
+					action.nameservers?.map( ( nameserver ) => ( {
+						nameserver,
+					} ) ) || [],
+			},
+		},
+		action
+	);
+
+export const updateDomainNameserversSuccess = ( { domainName }, nameservers ) => [
+	updateNameserversSuccess( domainName, nameservers ),
+	successNotice( translate( 'Yay, the name servers have been successfully updated!' ), {
+		duration: 5000,
+		id: `nameserver-update-notification-${ domainName }`,
+	} ),
+];
+
+export const updateDomainNameserversError = ( action, error ) =>
+	errorNotice( error.message, {
```

Comment:
Not sure how artificially I'm testing this but does it make sense to add a default message here? When I block the request URL on the client side it shows an empty error notice (because there's no response from the server). Not blocking ofc as it's the same behaviour as on the main branch.

Reply:
Sure, I think having a generic error message just in case won't hurt. Added in aa68af856650987203f13369b48e37ec6be858e7